src/NuGet.Core/NuGet.Commands/Internal/PackageSearchResultTable.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NuGet.Commands.Internal
+{
+    internal class PackageSearchResultTable
+    {
+        private readonly List<string[]> _rows = new List<string[]>();
+        private int[] _columnWidths;
+        private int[] _columnsToHighlight;
+
+        public PackageSearchResultTable(int[] columnsToHighlight, params string[] headers)
+        {
+            _columnsToHighlight = columnsToHighlight;
+            _columnWidths = new int[headers.Length];
+
+            for (int i = 0; i < headers.Length; i++)
+            {
+                _columnWidths[i] = headers[i].Length;
+            }
+
+            _rows.Add(headers);
+        }
+
+        public void AddRow(params string[] row)
+        {
+            if (row.Length != _columnWidths.Length)
+            {
+                throw new InvalidOperationException("Row column count does not match header column count.");
+            }
+
+            for (int i = 0; i < row.Length; i++)
+            {
+                _columnWidths[i] = Math.Max(_columnWidths[i], row[i]?.Length ?? 0);
+            }
+
+            _rows.Add(row);
+        }
+
+        public void PrintResult(string searchTerm = null)
+        {
+            ConsoleColor highlighterColor = GetHighlighterColor();
+
+            // If only headers are present (i.e., no package rows)
+            if (_rows.Count <= 1)
+            {
+                Console.WriteLine("No results found.");
+                return;
+            }
+
+            foreach (var row in _rows)
+            {
+                for (int i = 0; i < row.Length; i++)
+                {
+                    var paddedValue = (row[i] ?? string.Empty).PadRight(_columnWidths[i]);
+
+                    if (!string.IsNullOrEmpty(searchTerm) && paddedValue.IndexOf(searchTerm, StringComparison.OrdinalIgnoreCase) >= 0)
+                    {
+                        Console.Write("| ");
+                        if (_columnsToHighlight.Contains(i)) PrintWithHighlight(paddedValue, searchTerm, highlighterColor);
+                        else Console.Write(paddedValue);
+                        Console.Write(" ");
+                    }
+                    else
+                    {
+                        Console.Write("| " + paddedValue + " ");
```

Comment:
I'm the complete opposite of an accessibility expert. But while the formatted table looks nice to sighted people, I'm wondering what it sounds like to people who use screen readers. Have you tested?

Reply:
Looking into this. `dotnet tool search` also has a similar implementation

Reply:
@JonDouglas  reached out to a CLI expert about this concern. And the takeaway was that the table should be minimal, and it would be nice to have the option to export the output into an html or CSV format. I tried using the windows screen reader to tests it. It looks like the screen reader escapes the "|" and "-" characters I used for decorating the table, which is nice. However, on the other hand, it reads "=" used to separate package sources. As a result, I replaced them with "*", which it escaped. I am sure we can do better. And my idea is to add a `--format` option that would allow users to chose between a table, flattened list or a json.

Reply:
@Nigusu-Allehu is the default gonna remain table then?

Reply:
Yes, but I am not opposed to having the other options as default either.